packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {FacetContainer} from '../../../common/facets/facet-container/facet-container';
+import {FacetHeader} from '../../../common/facets/facet-header/facet-header';
+import {FacetValueCheckbox} from '../../../common/facets/facet-value-checkbox/facet-value-checkbox';
+import {FacetValueLabelHighlight} from '../../../common/facets/facet-value-label-highlight/facet-value-label-highlight';
+import {FacetValuesGroup} from '../../../common/facets/facet-values-group/facet-values-group';
+import {Hidden} from '../../../common/hidden';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+
+/**
+ * An automatic facet is a facet generated by the automatic facets feature. This component should
+ * not be used directly. It is internally used by the facet manager to automatically render updated
+ * automatic facets based on the search results.
+ *
+ * @part facet - The wrapper for the entire facet.
+ *
+ * @part label-button - The button that displays the label and allows to expand/collapse the facet.
+ * @part label-button-icon - The label button icon.
+ * @part clear-button - The button that resets the actively selected facet values.
+ * @part clear-button-icon - The clear button icon.
+ *
+ * @part values - The facet values container.
+ * @part value-label - The facet value label, common for all displays.
+ * @part value-count - The facet value count, common for all displays.
+ *
+ * @part value-checkbox - The facet value checkbox, available when display is 'checkbox'.
+ * @part value-checkbox-checked - The checked facet value checkbox, available when display is 'checkbox'.
+ * @part value-checkbox-label - The facet value checkbox clickable label, available when display is 'checkbox'.
+ * @part value-checkbox-icon - The facet value checkbox icon, available when display is 'checkbox'.
+ */
+@Component({
+  tag: 'atomic-automatic-facet',
+  styleUrl: 'atomic-automatic-facet.pcss',
+  shadow: true,
+})
+export class AtomicAutomaticFacet implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @State() public error!: Error;
+
+  @Prop({reflect: true}) public field!: string;
+  @Prop({reflect: true}) public facetId!: string;
+  @Prop({reflect: true}) public facet!: AutomaticFacet;
+  @Prop({reflect: true}) public searchStatus!: SearchStatus;
+  @Prop({reflect: true, mutable: true}) public isCollapsed = false;
+
+  @FocusTarget()
+  private headerFocus!: FocusTargetController;
+
+  private get numberOfSelectedValues() {
+    return this.facet.state.values.filter(({state}) => state === 'selected')
+      .length;
+  }
+
+  private renderValue(facetValue: FacetValue, onClick: () => void) {
+    const displayValue = getFieldValueCaption(
+      this.facet.state.field,
+      facetValue.value,
+      this.bindings.i18n
+    );
+    const isSelected = facetValue.state === 'selected';
```

Comment:
... and there, you essentially do the same thing.
Maybe you should refactor isSelected as a function:
```ts
const isSelected = (facetValue: FacetValue) => facetValue.state === 'selected';
```